packages/grid/x-data-grid/src/hooks/features/filter/gridFilterUtils.ts

```diff
@@ -263,36 +259,37 @@ const buildAggregatedFilterItemsApplier = (
 
   // We generate a new function with `eval()` to avoid expensive patterns for JS engines
   // such as a dynamic object assignment, e.g. `{ [dynamicKey]: value }`.
-  const filterItemTemplate = `(function filterItem$$(getRowId, appliers, row, shouldApplyFilter) {
-      ${appliers
-        .map(
-          (applier, i) =>
-            `const shouldApply${i} = !shouldApplyFilter || shouldApplyFilter(${JSON.stringify(
-              applier.item.field,
-            )});`,
-        )
-        .join('\n')}
-
-      const result$$ = {
-      ${appliers
-        .map(
-          (applier, i) =>
-            `${JSON.stringify(String(applier.item.id))}:
-          !shouldApply${i} ?
-            false :
-            appliers[${i}].fn(row),
-      `,
-        )
-        .join('\n')}};
-
-      return result$$;
-    })`;
-
-  const filterItemCore = evalCode<Function>(
+  const filterItemTemplate = `"use strict";
+${appliers
+  .map(
+    (applier, i) =>
+      `const shouldApply${i} = !shouldApplyFilter || shouldApplyFilter(${JSON.stringify(
+        applier.item.field,
+      )});`,
+  )
+  .join('\n')}
+
+const result$$ = {
+${appliers
+  .map(
+    (applier, i) =>
+      `  ${JSON.stringify(
+        String(applier.item.id),
+      )}: !shouldApply${i} ? false : appliers[${i}].fn(row),`,
+  )
+  .join('\n')}
+};
+
+return result$$;`;
+
+  const filterItemCore = new Function(
+    'appliers',
+    'row',
+    'shouldApplyFilter',
```

Comment:
I must say I don't like this version more because the parameters are listed at the `new Function()` call instead of at the top like they were before. The previous version feels a bit more readable.

As mentioned above, it's also already in strict mode because we don't call eval directly, so I don't see benefits to using `new Function`.

Reply:
Right, fair point. I have added a new commit to fix the readability regression.